Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -267,16 +270,23 @@ AppDelegateViewModelOutputs {
         .filter { $0 == .authorized }
         .take(first: 1)
         .mapConst(true)
-      let authorizationCompletedAndGranted = self.notificationAuthorizationCompletedProperty
-        .signal.filter(isTrue)
 
-      self.registerForRemoteNotifications = Signal.merge(
-        authorizationCompletedAndGranted,
-        userHasAuthorizedRemoteNotifications)
+      let authorizationCompletedAndGranted = self.notificationAuthorizationCompletedProperty.signal
+        .filter(isTrue)
+
+      self.registerForRemoteNotifications = authorizationCompletedAndGranted
         .ignoreValues()
+
+      // We always re-authorize to ensure that clients that had previously allowed only .alert
+      // are again authorized to allow [.alert, .badge]
+      self.authorizeForRemoteNotifications = Signal.merge(
+        self.didAcceptReceivingRemoteNotificationsProperty.signal,
+        userHasAuthorizedRemoteNotifications.ignoreValues()
+      )
     } else {
       //Never firing signal in ios 9
       self.getNotificationAuthorizationStatus = .empty
+      self.authorizeForRemoteNotifications = .empty
 
       self.registerForRemoteNotifications = applicationIsReadyForRegisteringNotifications
```

Comment:
If I'm reading things correctly here - we currently request push permissions on app start if the user if on iOS 9. This signal does not appear to take into account whether the alert was ever shown?